core/src/main/java/io/th0rgal/oraxen/utils/breaker/BreakerManager.java

```diff
@@ -89,7 +89,7 @@ private BukkitTask createBreakScheduler(double blockBreakTime, UUID breakerUUID)
                 activeBreakerData.sendBreakProgress();
 
                 for (ActiveBreakerData alterBreakerData : activeBreakerDataMap.values()) {
-                    if (alterBreakerData.breaker.getUniqueId().equals(breakerUUID)) continue;
+                    if (!alterBreakerData.breaker.getUniqueId().equals(breakerUUID)) continue;
```

Comment:
Idea here is that it would cancel all other tasks of other players that might be breaking the block
The stop digging packet might handle this but im not sure it does
Alternatively I think that handling it via OraxenBlockBreakEvent is possible here now that it correctly handles the timings

Reply:
```java
            } else if (EventUtils.callEvent(new BlockBreakEvent(block, player)) && ProtectionLib.canBreak(player, block.getLocation())) {
                PacketHelpers.removeMiningFatigue(player);
                activeBreakerData.resetProgress();
                activeBreakerData.sendBreakProgress();
```

instead of adding a OraxenBlockBreakEvent listener, we can just do it like this no?

Reply:
might be enough though theres alot of edge-cases im worrying about
in theory all this should be handled by the client as when the block is removed they should "abort" damaging the block and that event should trigger